File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.java
class Solution {
    public int[] solution(long n) {
        StringBuilder sb = new StringBuilder(Long.toString(n)).reverse();
        int[] answer = new int[sb.length()];
        
        for(int i = 0; i < answer.length; i++){
            answer[i] = sb.charAt(i) - '0';
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12932. 자연수 뒤집어 배열로 만들기/자연수 뒤집어 배열로 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(long long n) {
    int* answer = (int*)malloc(11 * sizeof(int));
    int idx = 0;
    
    while(n != 0){
        answer[idx++] = n % 10;
        n /= 10;
    }
    
    return answer;
}